astropy/utils/masked/tests/test_masked.py

```diff
@@ -1458,3 +1462,25 @@ class TestMaskedQuantityInteractionWithNumpyMA(
     TestMaskedArrayInteractionWithNumpyMA, QuantitySetup
 ):
     pass
+
+
+@pytest.mark.skipif(not HAS_PLT, reason="requires matplotlib.pyplot")
+@pytest.mark.xfail(MPL_LT_3_8, reason="this was fixed in matplotlib 3.8")
```

Comment:
Unless there is a sudden chance that it would xpass in `matplotlib<3.8` and we have to do something about that, maybe simpler to add this second check into the first skipif and only have one pytest marker applied?

Reply:
I was sure that `xfail` was suggested in a previous review but it doesn't look like it. I also didn't realise that `XPASS`es were configured to be treated as failures, and it's not desirable here so I'm happy to merge both decorators into a single `skipif`.

Reply:
Ah, actually it was suggested by @dhomeier in https://github.com/astropy/astropy/issues/12481#issuecomment-1892319898
But I don't think it should be a *strict* xfail, and a single `skipif` seems simpler than `skipif + xfail(strict=False)` 🤷🏻‍♂️

Reply:
@dhomeier , no need to overcomplicate this, right? Can we just have one skipif to test both conditions?